File: Benchmarking/30 Euler/euler9/euler9.java
import java.util.*;

public class euler9
{
	public static void main(String[] args)
	{
		long number = Long.parseLong(args[0]);
        pytha(number);
	}
	
	private static long pytha(long number)
	{
		HashMap<String, Long> dicto = new HashMap<>();
		long range = (long) Math.floor(number / 2);
		for(long a = 1; a < range; a++)
		{
			for(long b = a + 1; b < range; b++)
			{
				long c = number - a - b;
				if(a * a + b * b == c * c)
				{
					dicto.put("a", a);
					dicto.put("b", b);
					dicto.put("c", c);
				}
			}
		}
		
		long x = dicto.get("a") * dicto.get("b") * dicto.get("c");
		return x;
	}
}

File: Project-Euler/Stats/euler9/euler9.py
import sys
import math

number = int(sys.argv[1])
def pytha(number):
	dicto = {}
	for a in xrange(1, int(math.floor(number/3))):
		for b in xrange(a + 1, int(math.floor(number / 2))):
			c = number - a - b
			if a * a + b * b == c * c:
				dicto["a"] = a
				dicto["b"] = b
				dicto["c"] = c
	x = dicto["a"] * dicto["b"] * dicto["c"]
	return int(x)
	
print pytha(number)
